### ISSUE TITLE
Warning: Error in [[: subscript out of bounds

### ISSUE THREAD

**Issue by tommyhechina** [metaOmics/metaOmics#58]
Hellow! Here come with a problem when I use example data leukima for metaQC and metaDE. When I use metaQC, the program just stuck in the running phase for more than a hour. I check the R history and find something goes wrong. The warning message is listed below. 

![metaqc](https://user-images.githubusercontent.com/39785983/41103085-e47c6f22-6a9a-11e8-9ce1-ebb3f363260b.png)

```
Loading required package: PMA
Warning in data(pathway) : data set ‘pathway’ not found
Warning: Error in [[: subscript out of bounds
  75: unlist [server/meta_qc_server.R#24]
  73: getOption [server/meta_qc_server.R#24]
  72: observeEventHandler [server/meta_qc_server.R#74]
   1: runApp
```

When I try to use metaDE to analyze the example data leukima, the same problem occurs. This time the interface gives a warning message of subscipt out of bounds and the analysis cannot be runned. I don't know how to fix the problem and cannot find any information in the Google. Could you help me? Thanks! 
Warning message of metaDE is listed below.

![metade](https://user-images.githubusercontent.com/39785983/41103111-f5658fe4-6a9a-11e8-8de8-63cb36d142c9.png)

```
Warning: Error in [[: subscript out of bounds
  92: renderUI [server/meta_de_server.R#394]
  91: func
  78: origRenderFunc
  77: output$meta_de-covariate.opt
   1: runApp
```

**Comment by esradursunn** [comment]
Hi Tommy,

I've encountered exactly same problem. Did you solve this problem and what did you do for it? In particular, what did you utilize to cope with this problem or do you have any suggestion like using another package or tool for gene expresssion meta-analysis? I was very excited to see that the problem was the same, and I was disappointed when I saw there was no answer. I hope you can solve the problem because I really want to use this MetaOmics pipeline. 

Thanks you in advance.

**Comment by mzcs7** [comment]
I have the same problem.

**Comment by mzcs7** [comment]
@Caleb-Huo @esradursunn  @schwannden

**Comment by tommyhechina** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Hi mzcs7,
        Actually, I haven't figure out this problem and I abandon this method. In the first half year of 2018, I have asked the maintained of metaOmics and they could not find out what's wrong. But they gave a suggestion that I can download metaDE package or other package with old version and performed meta analysis by them. Though out of date and user-unfriendly, it still works. However, I abandon this method and find a better one whose name is NetworkAnalyst(https://www.networkanalyst.ca/). In this website you can also conduct a meta-analysis and do other analysis as well, not need to use metaOmics. Actually, I use neither of them and choose to perform meta-analysis in R by RobustRankAggred which suits my research.Hope this message can help you. And please inform me if you find out why this error occurs or how to fix it.Sincerely yours.
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                何懿Tommy
                            
                    
                        
                            
                                    tommyhechina@gmail.com
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 

    On 3/17/2019 11:53，mzcs7<notifications@github.com> wrote： 


@Caleb-Huo @esradursunn  @schwannden

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by matianzhou** [comment]
@tommyhechina @esradursunn @mzcs7 Sorry for the late reply (did not happen to see the message). Thanks so much for the questions. One way I can suggest at this moment is if you are familiar with R, you can run our stand-alone R packages MetaQC (https://github.com/metaOmics/MetaQC), MetaDE (https://github.com/metaOmics/MetaDE), MetaPath (https://github.com/metaOmics/MetaPath), etc. under metaOmics project. The stand-alone packages should be free of errors caused by version change in R or Shiny or Bioconductor. The test datasets worked well in all our maintainers' laptop and desktop (@Caleb-Huo @schwannden ) under both R 3.3 and 3.4, thus we never encountered such issues. But our team will work together to solve these issues and get back to you as soon as possible. Thanks again.

**Comment by matianzhou** [comment]
@tommyhechina @esradursunn @mzcs7 I just tried out the stand-alone R packages MetaQC and MetaDE in R 3.5 (install them by calling install_github() ) for the leukemia test data and they worked pretty well without any errors. For MetaDE, you can follow the example code in ?MetaDE; for MetaQC, please run the following to test leukemia dataset (the example dataset given in ?MetaQC is prostate cancer dataset):

library(MetaQC)
library(MetaDE)
data('Leukemia')
data('LeukemiaLabel')
data(pathway)
data <- Leukemia
select.group <- c('inv(16)','t(15;17)')
K <- length(data)
DList = colLabel= vector("list",K) ## for MetaQC, label has to be changed to 0/1 
for(k in 1:length(label)){
   k.index <- label[[k]] %in% select.group
   DList[[k]] <- data[[k]][,k.index]
   k.label <- label[[k]][k.index]
   k.label[k.label==select.group[1]] <- 0
   k.label[k.label==select.group[2]] <- 1
   colLabel[[k]] <- as.numeric(k.label)
}
GList=pathway[[1]]
filterGenes=TRUE
cutRatioByMean=0.3
cutRatioByVar=0.3
QCresult=MetaQC(DList, colLabel, GList,filterGenes,cutRatioByMean,cutRatioByVar)
plotMetaQC(QCresult$scoreTable)

Thanks. And we will continue to solve the issues in the Shiny application (first reproduce the errors you have encountered).

**Comment by matianzhou** [comment]
@tommyhechina @esradursunn @mzcs7 Both me and another maintainer (@Caleb-Huo) re-cloned the MetaOmics shiny from github and tried out in R 3.5.1, both MetaQC and MetaDE worked well without any errors (see figures below). For MetaQC, I would suggest you use the option "Perform gene filtering" =Yes, so the running time will be greatly reduced. We will try out in a Windows system some time next week to see if we can reproduce the errors you guys have. 

![Screen Shot 2019-03-17 at 10 43 29 AM](https://user-images.githubusercontent.com/15972932/54493704-63a5e480-48a9-11e9-810a-6968ce79eddc.png)
![Screen Shot 2019-03-17 at 10 45 34 AM](https://user-images.githubusercontent.com/15972932/54493705-63a5e480-48a9-11e9-91cc-79eceeb106ae.png)

**Comment by mzcs7** [comment]
@tommyhechina 
Tommy, thanks for your advice. But I need a strict quality assessment on my meta-analysis datasets. Anything else suggestions?

**Comment by mzcs7** [comment]
@matianzhou 
Thanks for your reply. From my point of view, I am really want to make quality control with the package, MetaQC. However, I noticed that there are only some picture or source codes about it in GitHub. Maybe you can support me with another detailed instruction file about the MetaQC's practical operation?
Thanks again!

**Comment by matianzhou** [comment]
@mzcs7 Of course, please find the MetaQC package manual and the original paper in the following. The MetaOmics tutorial (https://github.com/metaOmics/tutorial/blob/master/metaOmics_turtorial.pdf) also includes an introduction to the implementation of MetaQC, though in shiny, the options are pretty much the same in the package (in the stand-alone package, only one function needs to be called to implement MetaQC). You can follow the manual and tutorial to run it (the paper will give you more details why we proposed those assessment measures and how they are defined). If you still have specific questions on how to apply it to your data, I can ask the author of MetaQC to help. 

[MetaQC-manual.pdf](https://github.com/metaOmics/metaOmics/files/2978386/MetaQC-manual.pdf)
[Uploading MetaQC.pdf…]()

**Comment by mzcs7** [comment]
@matianzhou 
Thanks very much! I'll follow these tutorials firstly. And I will give you my feedbacks in time once I got some troubles. Really thanks for your precious time.
Wish you have a wonderful day.

**Comment by zyuuu7** [comment]
@matianzhou I've got the same question 'subscript out of bound' when I wanted to merge all my datasets. My R is 3.5.0 with windows. Have you already fixed it? And I don't know how to use metaOmics packages seperately. Can you show me the code? Thank you very much.

**Comment by Kingarrival27** [comment]
Hello all I got the same error in ubuntu 18.04  
```
Loading required package: sparcl
Loading required package: PMA
Warning: Error in [[: subscript out of bounds
  75: unlist [server/meta_qc_server.R#24]
  73: getOption [server/meta_qc_server.R#24]
  72: observeEventHandler [server/meta_qc_server.R#74]
   1: shiny::runApp
```
I have attached all the necessary packages for the metaomcis study.can us tell me where i missing them at specific

**Comment by Kingarrival27** [comment]
@Caleb-Huo @esradursunn  @schwannden

**Comment by matianzhou** [comment]
> Hello all I got the same error in ubuntu 18.04
> 
> ```
> Loading required package: sparcl
> Loading required package: PMA
> Warning: Error in [[: subscript out of bounds
>   75: unlist [server/meta_qc_server.R#24]
>   73: getOption [server/meta_qc_server.R#24]
>   72: observeEventHandler [server/meta_qc_server.R#74]
>    1: shiny::runApp
> ```
> 
> I have attached all the necessary packages for the metaomcis study.can us tell me where i missing them at specific

We have not tested on ubuntus before. Can you see my posts before? The test dataset should be error free in both windows and Mac (you may run MetaOmics in R instead of R studio). If you still encounter any issues, I would suggest you use the stand-alone packages (listed here: https://github.com/metaOmics), they all have concrete example code for you to test run. There is something we can hardly control about running GUI application on different types and versions of OS. Thanks.
